### PR TITLE
Fix crafting for fences etc. starting with colon

### LINKED ISSUE #3127: Registering fence/fence rail with a name starting in : creates node but no recipe
So if I register a fence using default it wont register recipe if fence starts with : node appears fine otherwise.
Fence gates work fine with names starting with : however they then have an extra : so its best to use any namespace initially. 

gate registers node like: minetest.register_node(":" .. name .. "_closed", fence_closed)
fence registers node like: minetest.register_node(name, def)

gate registers recipe after node, fence registers recipe before node.
both use name as output

Heres some code to test this yourself:

```
local text = "default_stone.png"
local matt = "default:stone"
default.register_fence(
            ":testmod:stone_fence",
            {
                description = "stone fence",
                texture = text,
                inventory_image = "default_fence_overlay.png^".. text .. "^default_fence_overlay.png^[makealpha:255,126,126",
                wield_image = "default_fence_overlay.png^".. text .. "^default_fence_overlay.png^[makealpha:255,126,126",
                material = matt,
                groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2},
                sounds = xcompat.sounds.node_sound_wood_defaults()
            }
            )

```
This works but there is no recipe. If you change ":testmod:stone_fence" to "testmod:stone_fence" the recipe works again.

Changing fence to match gate and then removing the : from the name works oddly enough. However I don't know if this might have unintended side effects.

Registering recipe after separately without : works aswel, when fence name starts with : and no changes to default. I wonder if some check to register recipe later if name starts with : could work? Still might register recipe twice if anyone else is doing this.

### PR DESCRIPTION
**Pull request by Emojigit**
Fixes #3127

This PR fixes the crafting recipes of fences, fence rails, and mese post lights. After the changes, even if the node name starts with a colon, the crafting recipe still exists.

This PR is ready for review.

### CODE REVIEW

File: mods/default/functions.lua [hunk 1]
```diff
@@ -364,10 +356,24 @@ function default.register_fence(name, def)
...
 	def.groups.fence = 1
 
+	local material = def.material
 	def.texture = nil
 	def.material = nil
 
 	minetest.register_node(name, def)
+
+	-- Register crafting recipe, trim away starting colon if any
+	if not material then return end
```
Comment (sfan5) [inline mods/default/functions.lua:366]: why would this be nil?

File: mods/default/functions.lua [hunk 2]
```diff
@@ -364,10 +356,24 @@ function default.register_fence(name, def)
...
+	if string.sub(name, 1,1) == ":" then
+		name = string.sub(name, 2)
+	end
```
Comment (sfan5) [inline mods/default/functions.lua:369]: `name:gsub("^:, "")` or similar should do (please double check)

Comment (Emojigit) [reply mods/default/functions.lua]: Done in 23104f2

Though the if-sub pattern should be faster, I agree that code tidiness is more important than the effectiveness pre-globalstep.

File: mods/default/functions.lua (hunk 1 above)
Comment (sfan5) [reply mods/default/functions.lua:366]: Makes sense, just document it in `game_api.txt` too.

Comment (Emojigit) [reply mods/default/functions.lua:366]: Done in 3b39117

Weirdly only fence is listed in that API documentation, so only that one is fixed.

File: mods/default/functions.lua (hunk 2 above)
Comment (appgurueu) [reply mods/default/functions.lua]: Personally I would've preferred `name:match(":?(.*)")` ;)

**Review by sfan5 — APPROVED** [review]

**Pull request merged**